lib/mail_room/connection.rb

```diff
@@ -50,6 +50,7 @@ def wait
 
         process_mailbox
       rescue Net::IMAP::Error, IOError
+        MailRoom.structured_logger.warn({ context: @mailbox.context, action: "Disconnected. Resetting..." })
```

Comment:
Should have access to `@mailbox` here.

Reply:
I've updated it to use the mailbox's logger instead.